lib/src/lints/avoid_final_with_getter/visitors/avoid_final_with_getter_visitor.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:solid_lints/src/lints/avoid_final_with_getter/visitors/getter_variable_visitor.dart';
+
+/// A visitor that checks for final private fields with getters.
+/// If a final private field has a getter, it is considered as a public field.
+class AvoidFinalWithGetterVisitor extends RecursiveAstVisitor<void> {
+  final _getters = <MethodDeclaration>[];
+
+  /// List of getters
+  Iterable<MethodDeclaration> get getters => _getters;
+
+  @override
+  void visitMethodDeclaration(MethodDeclaration node) {
+    if (node
+        case MethodDeclaration(
+          isGetter: true,
+          declaredElement: ExecutableElement(
+            isAbstract: false,
+            isStatic: false,
```

Comment:
Do we actually need to skip static getters? I think we may have the same situation when a static getter returns the static final private variable that either does not make any sense. I would remove this restriction.